lib/global_id/uri/gid.rb

```diff
@@ -83,7 +83,7 @@ def create(app, model, params = nil)
       def build(args)
         parts = Util.make_components_hash(self, args)
         parts[:host] = parts[:app]
-        parts[:path] = "/#{parts[:model_name]}/#{parts[:model_id]}"
+        parts[:path] = "/#{parts[:model_name]}/#{CGI.escape(parts[:model_id].to_s)}"
```

Comment:
@matthewd I considered whether it should be done with `:model_name` also. I'm ambivalent. Happy to add it if that's what you decide you want.